sdk/python/tests/unit/permissions/auth/server/test_auth_registry_server.py

```diff
@@ -65,33 +76,149 @@ def remote_feature_store(server_port, feature_store):
+        ALL_POSITIVE_TEST_NEEDED_PERMISSIONS, permissions
+    ):
+        entity_df = pd.DataFrame.from_dict(
+            {
+                # entity's join key -> entity values
+                "driver_id": [1001, 1002, 1003],
+                # "event_timestamp" (reserved key) -> timestamps
+                "event_timestamp": [
+                    datetime(2021, 4, 12, 10, 59, 42),
+                    datetime(2021, 4, 12, 8, 12, 10),
+                    datetime(2021, 4, 12, 16, 40, 26),
+                ],
+                # (optional) label name -> label values. Feast does not process these
+                "label_driver_reported_satisfaction": [1, 5, 3],
+                # values we're using for an on-demand transformation
+                "val_to_add": [1, 2, 3],
+                "val_to_add_2": [10, 20, 30],
+            }
+        )
 
-    assertpy.assert_that(entities).is_not_none()
-    assertpy.assert_that(len(entities)).is_equal_to(1)
+        training_df = client_fs.get_historical_features(
+            entity_df=entity_df,
+            features=[
+                "driver_hourly_stats:conv_rate",
+                "driver_hourly_stats:acc_rate",
+                "driver_hourly_stats:avg_daily_trips",
+                "transformed_conv_rate:conv_rate_plus_val1",
+                "transformed_conv_rate:conv_rate_plus_val2",
+            ],
+        ).to_df()
+        assertpy.assert_that(training_df).is_not_none()
+
+
+def _test_list_entities(
+    client_fs: FeatureStore, permissions: Union[Permission, list[Permission]]
+):
+    entities = client_fs.list_entities()
+
+    if not _is_auth_enabled(client_fs) or (
+        _is_auth_enabled(client_fs)
+        and (
+            _no_permission_retrieved(permissions)
+            or (
+                _is_listing_permissions_allowed(permissions)
+                and _permissions_exist_in_permission_list(
+                    [list_permissions_perm, list_entities_perm], permissions
+                )
+            )
+        )
+    ):
+        assertpy.assert_that(entities).is_not_none()
+        assertpy.assert_that(len(entities)).is_equal_to(1)
+        assertpy.assert_that(entities[0].name).is_equal_to("driver")
+    else:
+        assertpy.assert_that(entities).is_not_none()
+        assertpy.assert_that(len(entities)).is_equal_to(0)
+
+
+def _no_permission_retrieved(permissions: list[Permission]) -> bool:
+    return len(permissions) == 0
+
+
+def _test_list_permissions(
+    client_fs: FeatureStore, applied_permissions: Union[Permission, list[Permission]]
+) -> list[Permission]:
+    permissions = client_fs.list_permissions()
+
+    if not _is_auth_enabled(client_fs):
+        assertpy.assert_that(permissions).is_not_none()
+        assertpy.assert_that(len(permissions)).is_equal_to(len(applied_permissions))
+    elif _is_auth_enabled(client_fs) and _permissions_exist_in_permission_list(
+        ALL_POSITIVE_TEST_NEEDED_PERMISSIONS, permissions
```

Comment:
can this be generalized and avoid a specific validation for the `ALL_POSITIVE_TEST_NEEDED_PERMISSIONS` case?

Reply:
Changed it to [
    list_entities_perm,
    list_permissions_perm,
    list_fv_perm,
    list_odfv__perm,
    list_sfv__perm,
]
I hope this is what you ment